stdlib/public/core/Result.swift

```diff
@@ -0,0 +1,79 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+@_frozen
+public enum Result<Value, Error> {
+    case success(Value)
+    case failure(Error)
+    
+    var value: Value? {
+        switch self {
+        case let .success(value): return value
+        case .failure: return nil
+        }
+    }
+    
+    var error: Error? {
+        switch self {
+        case let .failure(error): return error
+        case .success: return nil
+        }
+    }
+    
+    var isSuccess: Bool {
+        switch self {
+        case .success: return true
+        case .failure: return false
+        }
+    }
+    
+    @_transparent
+    public init(_ throwing: () throws -> Value) {
```

Comment:
I was having trouble getting it to compile in the extension, but that was probably due to my initial build and test difficulties. I'll try again.

Reply:
Also, wouldn't I need to implement it in both `where Error == Swift.Error` and `where Error: Swift.Error` like `unwrap()`?

Reply:
Implementing for `where Error: Swift.Error` is problematic because the passed closure can throw an arbitrary `Swift.Error` value that is not necessarily the same type as the `Error` placeholder. You'd therefore run into much the same issue where you'd have to force cast the thrown error to `Error`. Therefore IMO it should only be implemented for `where Error == Swift.Error`.